org.eclipse.scout.rt.server.commons/src/main/java/org/eclipse/scout/rt/server/commons/servlet/HttpProxy.java

```diff
@@ -240,21 +238,20 @@ protected HttpContext createHttpContext() {
    * Rewrites the <code>pathInfo</code> part of the current request if the rewriteRule and rewriteReplacement is set on
    * the options object. This allows to redirect the request to a different URL than the URL that has been requested.
    */
-  protected URI rewriteUrl(HttpServletRequest req, HttpProxyRequestOptions options) {
-    String pathInfo = req.getPathInfo();
+  protected String rewriteUrl(HttpServletRequest req, HttpProxyRequestOptions options) {
+    String pathInfo = ObjectUtility.nvl(req.getPathInfo(), "");
     IRewriteRule rewriteRule = options.getRewriteRule();
     if (rewriteRule != null) {
       pathInfo = rewriteRule.rewrite(pathInfo);
     }
-    UriBuilder uriBuilder = new UriBuilder(getRemoteBaseUrl());
-    if (StringUtility.startsWith(pathInfo, "/")) {
-      // addPath on UriBuilder adds a beginning slash on its own; strip our beginning slash to avoid duplicated slashes
+    // pathInfo must be url-encoded; except for forward-slashes they should be kept
+    pathInfo = IOUtility.urlEncode(pathInfo).replaceAll("%2F", "/");
+    String remoteBaseUrl = getRemoteBaseUrl();
+    if (remoteBaseUrl.endsWith("/") && pathInfo.startsWith("/")) {
```

Comment:
As discussed, check both conditions separately and remove the / then.
String url = remoteBaseUrl + "/" + pathInfo;
Use StringUtility.join("?", url, req.getQueryString())

Reply:
* Removed trailing slash check for remoteBaseUrl completely, according to javadoc on getRemoteBaseUrl it should never include a trailing slash (added however a trailing slash check in withRemoteBaseUrl).
* An URL with an empty pathInfo (and w/o) slash may be valid after all, e.g. set remoteBaseUrl for a specific ressource remoteBaseUrl = http://my-server/my-servlet - in this case one may want to create a proxy accessing this specific resource w/o any slashes.